Rust/cargo-azsphere/src/config/mod.rs

```diff
@@ -0,0 +1,159 @@
+use cargo_toml::Error as CargoTomlError;
+use cargo_toml::Manifest;
+use std::path::{Path, PathBuf};
+use toml::value::Value;
+
+use crate::error::{ConfigError, Error};
+use metadata::{CompoundMetadataConfig, ExtraMetaData, MetadataConfig, TomlValueHelper};
+
+mod metadata;
+
+#[derive(Debug, Clone)]
+pub enum ExtraMetadataSource {
+    File(PathBuf, Option<String>),
+    Text(String),
+}
+
+#[derive(Debug)]
+pub struct Config {
+    manifest: Manifest,
+    _manifest_path: PathBuf,
+    extra_metadata: Vec<ExtraMetaData>,
+}
+
+/// Required fields, retrieved from the app's Cargo.toml
+#[derive(Debug)]
+pub struct PackageConfig {
+    pub name: String, // Cargo.Toml package.name
```

Comment:
These comments would be better as doc comments

```suggestion
    /// Cargo.Toml package.name
    pub name: String, 
```